workflows/general/alignment-post.wdl

```diff
@@ -10,27 +14,27 @@ workflow alignment_post {
         bam: "Input BAM format file to process"
         mark_duplicates: "Add SAM flag to computationally determined duplicate reads?"
         contaminant_db: "A compressed reference database corresponding to the aligner chosen with `xenocp_aligner` for the contaminant genome"
-        cleanse_xenograft: "If true, use XenoCP to unmap reads from contaminant genome"
+        max_retries: "Number of times to retry failed steps. Overrides task level defaults."
         xenocp_aligner: {
             description: "Aligner to use to map reads to the host genome for detecting contamination"
             choices: [
-                'bwa aln',
-                'bwa mem',
-                'star'
+                'bwa aln',  # TODO this is SE. Is `bwa_aln_pe` a possibility?
+                'bwa mem',  # TODO this is also SE.
+                'star'  # TODO I think only this will work with paired end? What would DNA use?
```

Comment:
Can you should some light here @adthrasher ? Am I correct that these bwa tasks are for SE data?

Reply:
XenoCP converts reads to fastq, each mate in a pair gets a fastq record, and it then processes the reads as single-end. So each of the aligners here is run in single-end mode. All three can be run in paired-end mode for other applications, but XenoCP only uses the single-end mode.

In `workflows`, we have `bwa_aln` (single-end), `bwa_aln_pe` (paired-end), `bwa_mem` (single-end), and `star.alignment` (single-end or paired-end). I don't believe `bwa_aln_pe` is being used by any code at the moment. It was going to be part of the ChIP-Seq workflow until it was decided to use SEAseq.

Reply:
Well it's good to know that STAR SE mode is being tested 😅  TBH I wasn't positive it would run correctly, but I guess XenoCP has been using it just fine.

I think we need documentation somewhere for the differences between these aligners. Not sure _here_ is the right place, but we could use that `external_documentation` key to point to the relevant docs. Does that documentation already exist?

Reply:
Not any of which I am aware. What sort of documentation are you looking for though? Are you thinking in the tasks themselves? Or something in the XenoCP/workflows metadata to guide a choice?